src/Workspaces/Core/Portable/Workspace/Workspace.cs

```diff
@@ -1186,6 +1188,21 @@ private void CheckAllowedProjectChanges(ProjectChanges projectChanges)
             }
         }
 
+        private bool CanApplyParseOptionChange(ParseOptions oldOptions, ParseOptions newOptions, Project project)
```

Comment:
@jcouv: don't mention the IParseOptionsService: that's an implementation detail. Just make this method protected virtual like CanApplyChange, and move the entire logic into the VS layer.